File: qrcodereaderview/src/main/java/com/dlazaro66/qrcodereaderview/SimpleLog.java
package com.dlazaro66.qrcodereaderview;

import android.util.Log;

public class SimpleLog {

  private static boolean loggingEnabled = false;

  public static void setLoggingEnabled(boolean enabled) {
      loggingEnabled = enabled;
  }

  public static void d(String tag, String text) {
      if (loggingEnabled) {
          Log.d(tag, text);
      }
  }

  public static void w(String tag, String text) {
      if (loggingEnabled) {
          Log.w(tag, text);
      }
  }

  public static void w(String tag, String text, Throwable e) {
      if (loggingEnabled) {
          Log.w(tag, text, e);
      }
  }

  public static void e(String tag, String text) {
      if (loggingEnabled) {
          Log.e(tag, text);
      }
  }

  public static void d(String tag, String text, Throwable e) {
      if (loggingEnabled) {
          Log.d(tag, text, e);
      }
  }

  public static void i(String tag, String text) {
      if (loggingEnabled) {
          Log.i(tag, text);
      }
  }
}

File: qrcodereaderview/src/main/java/com/dlazaro66/qrcodereaderview/SimpleLog.kt
package com.dlazaro66.qrcodereaderview

import android.util.Log

object SimpleLog {
    private var loggingEnabled = false
    fun setLoggingEnabled(enabled: Boolean) {
        loggingEnabled = enabled
    }

    fun d(tag: String?, text: String?) {
        if (loggingEnabled) {
            Log.d(tag, text ?: "")
        }
    }

    fun w(tag: String?, text: String?) {
        if (loggingEnabled) {
            Log.w(tag, text ?: "")
        }
    }

    @JvmStatic
    fun w(tag: String?, text: String?, e: Throwable?) {
        if (loggingEnabled) {
            Log.w(tag, text, e)
        }
    }

    fun e(tag: String?, text: String?) {
        if (loggingEnabled) {
            Log.e(tag, text ?: "")
        }
    }

    fun d(tag: String?, text: String?, e: Throwable?) {
        if (loggingEnabled) {
            Log.d(tag, text, e)
        }
    }

    @JvmStatic
    fun i(tag: String?, text: String?) {
        if (loggingEnabled) {
            Log.i(tag, text ?: "")
        }
    }
}